spec/board_spec.rb

```diff
@@ -32,4 +32,61 @@
       expect(@board.valid_coordinate?("A22")).to be  false
     end
   end
+
+  describe 'placment validation' do
+    it 'checks before ship placement' do
+      expect(board.valid_placement?(cruiser, ["A1", "A2"])).to be false
+      expect(board.valid_placement?(submarine, ["A2", "A3", "A4"])). to be false
+    end
+  
+    it 'has consecutive coordinates' do
+      expect(board.valid_placement?(cruiser, ["A1", "A2", "A4"])).to be false
+      expect(board.valid_placement?(submarine, ["A1", "C1"])).to be false
+      expect(board.valid_placement?(cruiser, ["A3", "A2", "A1"])).to be false
+      expect(board.valid_placement?(submarine, ["C1", "B1"])).to be false
+    end
+  
+    it 'is not diagonal' do
```

Comment:
`cruiser` and `submarine` set-up here or with `@` symbol in the `before(:each)` statement